src/gbm/gbtree_model.cc

```diff
@@ -10,18 +10,35 @@ namespace gbm {
 
 void GBTreeModel::Save(dmlc::Stream* fo) const {
   CHECK_EQ(param.num_trees, static_cast<int32_t>(trees.size()));
-  fo->Write(&param, sizeof(param));
+
+  if (DMLC_IO_NO_ENDIAN_SWAP) {
+    fo->Write(&param, sizeof(param));
+  } else {
+    auto x = param.ByteSwap();
+    fo->Write(&x, sizeof(x));
+  }
   for (const auto & tree : trees) {
     tree->Save(fo);
   }
   if (tree_info.size() != 0) {
-    fo->Write(dmlc::BeginPtr(tree_info), sizeof(int32_t) * tree_info.size());
+    if (DMLC_IO_NO_ENDIAN_SWAP) {
```

Comment:
Can we put the condition into `ByteSwap` function?  If true then `ByteSwap` is a no op.

Reply:
@trivialfis sorry but can you explain more about `no op`? The intuition here is that `ByteSwap` only needs to be invoked when the condition `DMLC_IO_NO_ENDIAN_SWAP` does not meet, thus it seems to me not that reasonable to put the condition into `ByeSwap`.

Reply:
Hi @trivialfis, would you mind me marking this as `resolved` if we agreed on leaving this condition outside of `ByteSwap`?